connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/aiagent/model/request/ProviderConfiguration.java

```diff
@@ -135,6 +142,160 @@ public record AnthropicModelParameters(
     }
   }
 
+  @TemplateSubType(id = AZURE_OPENAI_ID, label = "Azure OpenAI")
+  record AzureOpenAiProviderConfiguration(@Valid @NotNull AzureOpenAiConnection azureOpenAi)
+      implements ProviderConfiguration {
+
+    @TemplateProperty(ignore = true)
+    public static final String AZURE_OPENAI_ID = "azureOpenAi";
+
+    public record AzureOpenAiConnection(
+        @FEEL
+            @TemplateProperty(
+                group = "provider",
+                description =
+                    "Specify Azure OpenAI endpoint. Details in the <a href=\"https://learn.microsoft.com/en-us/azure/ai-foundry/openai/reference\" target=\"_blank\">documentation</a>.",
+                constraints = @PropertyConstraints(notEmpty = true))
+            String endpoint,
+        @Valid @NotNull AzureAuthentication authentication,
+        @Valid @NotNull AzureOpenAiModel model) {}
+
+    @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+    @JsonSubTypes({
+      @JsonSubTypes.Type(
+          value =
+              AzureOpenAiProviderConfiguration.AzureAuthentication.AzureApiKeyAuthentication.class,
+          name = "apiKey"),
+      @JsonSubTypes.Type(
+          value =
+              AzureOpenAiProviderConfiguration.AzureAuthentication
+                  .AzureClientCredentialsAuthentication.class,
+          name = "clientCredentials")
+    })
+    @TemplateDiscriminatorProperty(
+        label = "Authentication",
+        group = "provider",
+        name = "type",
+        defaultValue = "apiKey",
+        description =
+            "Specify the Azure OpenAI authentication strategy. Learn more at the <a href=\"https://docs.langchain4j.dev/integrations/language-models/azure-open-ai\" target=\"_blank\">documentation page</a>")
```

Comment:
🔧 Do the Langchain4J linked docs really contain helpful information about the authentication strategy? I would prefer if we could link our own docs here because that allows us to describe the auth strategies we support.